src/main/scala/com/auth/example/domain/User.scala

```diff
@@ -1,3 +1,7 @@
 package com.auth.example.domain
 
-final case class User(email: String, password: String)
+final case class Credentials(email: String, password: String)
+
+final case class DBUser(email: String, password: String)
+
+final case class AuthedUser(email: String, password: String)
```

Comment:
Took the liberty of changing the model to make it easier to reason about the various "stages" of a user, even though all of these are structurally the same.